agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4432,6 +4432,52 @@ bool slave_thread::handle_channel_preference_query(Socket *sd, ieee1905_1::CmduM
     return message_com::send_cmdu(ap_manager_socket, cmdu_tx);
 }
 
+/**
+ * @brief check if channel is restricted in a given preference
+ * 
+ * @param channel channel to check
+ * @param preference channel preference
+ * @return true if channel is restricted
+ * @return false if channel is not restricted
+ */
+static bool is_channel_restricted(const beerocks::message::sWifiChannel channel,
+                                  const son::wireless_utils::sChannelPreference &preference)
+{
+    uint8_t center_channel = 0;
+    auto bw                = static_cast<beerocks::eWiFiBandwidth>(channel.channel_bandwidth);
+    auto operating_class   = wireless_utils::get_operating_class_by_channel(channel);
+
+    // operating classes 128,129,130 use center channel **unlike the other classes**,
+    // so convert channel and bandwidth to center channel.
+    // For more info, refer to Table E-4 in the 802.11 specification.
+    if (operating_class == 128 || operating_class == 129 || operating_class == 130) {
+        center_channel = wireless_utils::get_5g_center_channel(channel.channel, bw, true);
+    }
+
+    // restrict channels from other operating classes
+    if (operating_class != preference.oper_class) {
+        return true;
+    }
```

Comment:
This comment makes little sense to me. I think you mean "consider channels in different operating classes as restricted".

Also I'd do this check more in the beginning.

nitpick both